### PR TITLE
Lose the ghc-lib flag

### LINKED ISSUE #784: Reenable -fghc-lib build
The ghc-lib build (using flag `ghc-lib` for the `ghcide` package) needs a test. https://github.com/haskell/haskell-language-server/commit/0db12a71 shows what is missing, but needs adjusting to build only ghcide and nothing else. Probably best to create a separate workflow for it

### PR DESCRIPTION
**Pull request by pepeiborra**
@cocreature confirmed that DA are not depending on this anymore and we are free to drop it. 

Closes #784

### CODE REVIEW

File: ghcide/src/Development/IDE/Spans/Documentation.hs [hunk 1]
```diff
@@ -71,23 +70,21 @@ lookupKind :: HscEnv -> Module -> Name -> IO (Maybe TyThing)
...
   case res of
-      Left _ -> mapM mkSpanDocText names
+      Left _ -> return []
       Right res -> zipWithM unwrap res names
   where
     unwrap (Right (Just docs, _)) n = SpanDocString docs <$> getUris n
     unwrap _ n = mkSpanDocText n
 
     mkSpanDocText name =
-      SpanDocText (getDocumentation sources name) <$> getUris name
```
Comment (wz1000) [inline ghcide/src/Development/IDE/Spans/Documentation.hs:90]: `getDocumentation` can also be deleted.

**Comment by wz1000** [comment]
There is a `ghc-lib` flag in `hie-compat` that can also be removed

File: ghcide/src/Development/IDE/Spans/Documentation.hs (hunk 1 above)
Comment (wz1000) [reply ghcide/src/Development/IDE/Spans/Documentation.hs:90]: In fact, most of this file below this function can be deleted.

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Spans/Documentation.hs:90]: No, `getDocumentation` is still used in the local completions code

**Comment by pepeiborra** [comment]
> There is a `ghc-lib` flag in `hie-compat` that can also be removed

That doesn't seem a huge priority since `hie-compat` is relatively stable

**Review by wz1000 — APPROVED** [review]

**Pull request merged**